yaptide/celery/tasks.py

```diff
@@ -0,0 +1,191 @@
+            raise celery_exceptions.TaskError
+
+        estimators_dict: dict = runner_obj.get_data()
+
+        result: dict = pymchelper_output_to_json(estimators_dict)
+
+        return {'status': 'COMPLETED', 'result': result}
+
+
+def pymchelper_output_to_json(estimators_dict: dict) -> dict:
+    """Dummy function for converting simulation output to dictionary"""
+    if not estimators_dict:
+        return {'message': 'No estimators'}
+
+    # result_dict is a dictionary, which is later converted to json
+    # to provide readable API response for fronted
+    # keys in results_dict are estimator names, values are the estimator objects
+    result_dict = {'estimators': []}
+    estimator: Estimator
+    for estimator_key, estimator in estimators_dict.items():
+        # est_dict contains list of pages
+        est_dict = {
+            'name': estimator_key,
+            'pages': [],
+            }
+
+        for page in estimator.pages:
+            # page_dict contains:
+            # 'dimensions' indicating it is 1 dim page
+            # 'data' which has unit, name and list of data values
+            page_dict = {
+                'dimensions': page.dimension,
+                'data': {
+                    'unit': str(page.unit),
+                    'name': str(page.name),
+                    'values': page.data_raw.tolist(),
+                }
+            }
+            # currently output is returned only when dimension == 1 due to
+            # problems in efficient testing of other dimensions
+
+            if page.dimension in {1, 2}:
+                axis: MeshAxis = page.plot_axis(0)
+                page_dict['first_axis'] = {
+                    'unit': str(axis.unit),
+                    'name': str(axis.name),
+                    'values': axis.data.tolist(),
+                }
+            if page.dimension == 2:
+                axis: MeshAxis = page.plot_axis(1)
+                page_dict['second_axis'] = {
+                    'unit': str(axis.unit),
+                    'name': str(axis.name),
+                    'values': axis.data.tolist(),
+                }
+            if page.dimension > 2:
+                # Add info about the location of the file containging to many dimensions
+                raise ValueError(f'Invalid number of pages {page.dimension}')
+
+            est_dict['pages'].append(page_dict)
+        result_dict['estimators'].append(est_dict)
+
+    return result_dict
+
+
+@celery_app.task
+def simulation_task_status(task_id: str) -> dict:
+    """Task responsible for returning simulation status"""
+    task = AsyncResult(id=task_id, app=celery_app)
+
+    result = {
+        'status': 'OK',
+        'message': {
+            'state': task.state
+        }
+    }
+    if task.state == "PENDING":
+        result['message']['status'] = 'Pending...'
+    elif task.state == "PROGRESS":
+        result['message']['status'] = 'Calculations in progress...'
```

Comment:
I think we have somewhere simulation type being defined.
Can you add here an if statement and call this method only if the simulation type is `shieldhit` ?

Reply:
changed